spec/src/modules/tasks.js

```diff
@@ -0,0 +1,179 @@
+/* eslint-disable no-unused-expressions, import/no-unresolved, no-restricted-syntax, max-nested-callbacks */
+const dotenv = require('dotenv');
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+const sinon = require('sinon');
+const sinonChai = require('sinon-chai');
+const nodeFetch = require('node-fetch').default;
+const cloneDeep = require('lodash.clonedeep');
+const ConstructorIO = require('../../../test/constructorio'); // eslint-disable-line import/extensions
+const helpers = require('../../mocha.helpers');
+
+chai.use(chaiAsPromised);
+chai.use(sinonChai);
+dotenv.config();
+
+const sendTimeout = 300;
+const testApiKey = process.env.TEST_API_KEY;
+const testApiToken = process.env.TEST_API_TOKEN;
+const validOptions = {
+  apiKey: testApiKey,
+  apiToken: testApiToken,
+};
+
+describe('ConstructorIO - Tasks', function ConstructorIOTasks() {
+  // Ensure Mocha doesn't time out waiting for operation to complete
+  this.timeout(10000);
+
+  const clientVersion = 'cio-mocha';
+  let fetchSpy;
+  let taskId;
+
+  before(async () => {
+    const { catalog } = new ConstructorIO({
+      ...validOptions,
+    });
+
+    // Grab items file from Integration Examples repo and upload
+    const itemsResponse = await nodeFetch('https://raw.githubusercontent.com/Constructor-io/integration-examples/main/catalog/items.csv');
+    const itemsBuffer = await itemsResponse.buffer();
+
+    const data = {
+      items: itemsBuffer,
+      section: 'Products',
+    };
+
+    // Save task id of uploaded catalog for the following tests below
+    await catalog.replaceCatalog(data).then((res) => {
+      taskId = res.task_id;
+    });
+  });
+
+  beforeEach(() => {
+    global.CLIENT_VERSION = clientVersion;
+    fetchSpy = sinon.spy(nodeFetch);
+  });
+
+  afterEach((done) => {
+    delete global.CLIENT_VERSIONS;
+
+    fetchSpy = null;
+
+    setTimeout(done, sendTimeout);
+  });
+
+  describe('getAllTasks', () => {
+    it('Should retrieve a list of tasks', (done) => {
+      const { tasks } = new ConstructorIO({
+        ...validOptions,
+        fetch: fetchSpy,
+      });
+
+      tasks.getAllTasks().then((res) => {
+        const requestedUrlParams = helpers.extractUrlParamsFromFetch(fetchSpy);
+
+        expect(fetchSpy).to.have.been.called;
+        expect(res.total_count).to.be.gte(1);
+        expect(res.tasks[0].id).to.eq(taskId);
```

Comment:
Should we check all items in `res.tasks`? Could break if multiple people happen to be running the tests at the same time.

Reply:
Updated, should look for the task that was uploaded before running the expect